hilo.rb

```diff
@@ -1,2 +1,22 @@
-number = rand(1..100)
-puts number 
\ No newline at end of file
+random_number = rand(1..100)
+puts "Enter a number between 1 and 100. press enter to submit your guess"
+input = gets.chomp
+guess = input.to_i
+response = ''
+count = 0
+while count < 5
+    count = count + 1
+    response = ''
+    if guess > 100
```

Comment:
Why not check first if its an whole number?
```
if guess.to_s != input
        response = "Not an integer"
```

Reply:
I don't quite understand how converting the guess back to a string and comparing to the original input will check if it's a whole number?

Reply:
Is it becasue to_i converts the string to 0 if it can't parse it as an actual number and then when you convert 0 back to a string and compare it to the original input it won't match - then you can assume that the original input was not a number?

Reply:
Based on the ruby docs it `.to_i` interprets the leading characters for a string as an integer base: https://ruby-doc.org/core-3.0.0/String.html#method-i-to_i.  What is quite clever is the left-hand side of the equation already determines if its a valid integer or not.  All the equation does is match it with the inputted string.

Reply:
oh ok, so .to_s will only convert a value if it's an integer?